tests/suite/nginxgateway_test.go

```diff
@@ -0,0 +1,180 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	core "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	ngfAPI "github.com/nginxinc/nginx-gateway-fabric/apis/v1alpha1"
+	"github.com/nginxinc/nginx-gateway-fabric/tests/framework"
+)
+
+var _ = Describe("NginxGateway", Ordered, Label("functional", "nginxGateway"), func() {
+	var (
+		ngfPodName string
+
+		namespace          = "nginx-gateway"
+		nginxGatewayNsname = types.NamespacedName{Name: releaseName + "-config", Namespace: namespace}
+
+		files = []string{
+			"nginxgateway/nginx-gateway.yaml",
+		}
+	)
+
+	AfterAll(func() {
+		// cleanup and restore NGF instance
+		teardown(releaseName)
+		setup(getDefaultSetupCfg())
+	})
+
+	getNginxGateway := func(nsname types.NamespacedName) (ngfAPI.NginxGateway, error) {
+		ctx, cancel := context.WithTimeout(context.Background(), timeoutConfig.GetTimeout)
+		defer cancel()
+
+		var nginxGateway ngfAPI.NginxGateway
+
+		if err := k8sClient.Get(ctx, nsname, &nginxGateway); err != nil {
+			return nginxGateway, err
+		}
+
+		return nginxGateway, nil
+	}
+
+	verifyAndReturnNginxGateway := func(nsname types.NamespacedName) ngfAPI.NginxGateway {
+		nginxGateway, err := getNginxGateway(nsname)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(nginxGateway).ToNot(BeNil())
+
+		Expect(nginxGateway.Status.Conditions).To(HaveLen(1))
+		condition := nginxGateway.Status.Conditions[0]
+
+		Expect(condition.Type).To(Equal("Valid"))
+		Expect(condition.Reason).To(Equal("Valid"))
+
+		return nginxGateway
+	}
+
+	getNGFPodName := func() (string, error) {
+		podNames, err := framework.GetReadyNGFPodNames(
+			k8sClient,
+			ngfNamespace,
+			releaseName,
+			timeoutConfig.GetTimeout,
+		)
+		if err != nil {
+			return "", err
+		}
+
+		if len(podNames) != 1 {
+			return "", fmt.Errorf("expected 1 pod name, got %d", len(podNames))
+		}
+
+		return podNames[0], nil
+	}
+
+	When("testing NGF on startup", func() {
+		When("log level is set to debug", func() {
+			It("outputs debug logs and the status is accepted and true", func() {
+				ngfPodName, err := getNGFPodName()
+				Expect(err).ToNot(HaveOccurred())
+
+				_ = verifyAndReturnNginxGateway(nginxGatewayNsname)
+
+				logs, err := resourceManager.GetPodLogs(ngfNamespace, ngfPodName, &core.PodLogOptions{
+					Container: "nginx-gateway",
+				})
+				Expect(err).ToNot(HaveOccurred())
+
+				Expect(logs).To(ContainSubstring("\"level\":\"debug\""))
```

Comment:
Has this check ever failed? Wondering if we should wrap this in a retry to avoid flakiness.

Reply:
Check has never failed, though I guess I can wrap the whole pod log checking in a retry as there would be no harm.